crawler/src/main/java/com/brandwatch/minibcr/crawler/model/reddit/Subreddit.java

```diff
@@ -0,0 +1,46 @@
+package com.brandwatch.minibcr.crawler.model.reddit;
+
+import java.util.List;
+
+public class Subreddit {
+
+    private Subreddit data;
+    private List<Subreddit> children;
```

Comment:
This class confuses me. What does it represent exactly?

Reply:
reddit crawler response class for the listing api. ( response for a particular subreddit )

Reply:
It's a class called `Subreddit` with a `Subreddit` field - is that intentional? I have never seen this and am highly confused that this even works. How is that non an infinite loop? Maybe I am learning something funny about Java here.

Reply:
Its like Node within a Node structure. At least that's how I assumed seeing the response. 
![image](https://user-images.githubusercontent.com/115222581/201313491-a323e043-b0a5-4f3d-88e5-e90d17ef0c43.png)

Reply:
Can you maybe show me an example response from the API as plain JSON?

Reply:
Is it like this?
https://www.reddit.com/r/phone/new.json

Reply:
Yes right.

Reply:
I continue to think that while this may somehow work, it's not an accurate representation of the API. It looks like because certain keywords are repeated, specifically `kind` and `data`, you assumed some kind of nesting of the same thing, but I don't think that's actually what's happening. The result is that you created some combined object that contains field like `data` and `children` but also `selftext` and `title` for example, which doesn't actually exist like that anywhere in the Reddit API.

The structure of the API is documented [here](https://www.jcchouinard.com/documentation-on-reddit-apis-json/) for example:

```json
{
    "kind": "string", 
    "data": {
        "modhash": "string", 
        "dist": int, 
        "children": [{
            "kind": "string", 
            "data": {
                "approved_at_utc":"string", 
                "subreddit": "string", 
                "selftext": "string" 
                ...,
                "is_video":"boolean"
            }],
        "after":"",
        "before:""
    }
}
```

So if you break it down, I think it's something closer to this. I represented it as records, but you could also do classes if you prefer:

```java
public record RedditApiResponse(
        String kind,
        RedditApiResponseData data
) {
    private record RedditApiResponseData(
            String modhash,
            int dist,
            String after,
            String before,
            List<RedditApiResponseDataChildren> children
            ...
    ) {
        private record RedditApiResponseDataChildren(
                String kind,
                List<RedditApiResponseDataChildrenData> children
        ) {
            private record RedditApiResponseDataChildrenData(
                    String subreddit,
                    String selftext,
                    String title,
                    ...
            ) {
            }
        }
    }
}
```

Obviously could could leave out or add fields as you need them. You might also opt against inner records/classes in favour of independent ones. You might also find much better names than I did (they're shit) 😄 

But I think this would represent the API more accurately.